compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -403,7 +403,8 @@ public boolean checkForUniqueMemberSymbol(Location pos, SymbolEnv env, BSymbol s
     public BSymbol resolveBinaryOperator(OperatorKind opKind,
                                          BType lhsType,
                                          BType rhsType) {
-        return resolveOperator(names.fromString(opKind.value()), Lists.of(lhsType, rhsType));
+        boolean isAdditiveExpr = opKind.equals(OperatorKind.ADD) || opKind.equals(OperatorKind.SUB);
+        return resolveOperator(isAdditiveExpr, names.fromString(opKind.value()), Lists.of(lhsType, rhsType));
```

Comment:
I think this flow will be more clear if
1. we pass `OperatorKind` instead of both `isAdditiveExpr` and `names.fromString(opKind.value())`
2. Use `OperatorKind` to identify whether it is additive-expr or not (at L:2158)
WDYT?

Reply:
By doing that we cannot identify whether it is a binary or unary operator.

Reply:
Why can't we evaluate `isAdditiveExpr` at the place where it is needed if we pass `OperatorKind` only?
Did I miss something?

Reply:
We can, But we need to check whether it is binary expr(`1+1`) or unary expr(+1) also. In L419 I have directly marked it as not an additive expr when the operator is unary. 
So if we pass only the operator kind we need to do a check(in L:2158) to identify whether it is a binary or unary operator. right?

Reply:
hmm.. I understood. I think we have to handle the binary cases and the unary cases separately. Let's discuss and do in seperately.